torchvision/transforms/functional_tensor.py

```diff
@@ -917,3 +940,70 @@ def perspective(
     mode = _interpolation_modes[interpolation]
 
     return _apply_grid_transform(img, grid, mode)
+
+
+def _get_kernel(radius: float, passes: int):
+    sigma2 = torch.Tensor([radius ** 2 / passes])
+
+    kernel_rad = (torch.sqrt(12. * sigma2 + 1.) - 1.) / 2.
+
+    kernel_rad_int = kernel_rad.long().item()
+
+    kernel_rad_float = (2 * kernel_rad_int + 1) * (kernel_rad_int * (kernel_rad_int + 1) - 3 * sigma2)
+    kernel_rad_float /= 6 * (sigma2 - (kernel_rad_int + 1) * (kernel_rad_int + 1))
+    kernel_rad_float = kernel_rad_float.item()
+
+    kernel_rad = kernel_rad_int + kernel_rad_float
+
+    ksize = 2 * kernel_rad_int + 1 + 2 * (kernel_rad_float > 0)
+    kernel1d = torch.ones(ksize) / (2 * kernel_rad + 1)
+
+    if kernel_rad_float > 0:
+        kernel1d[[0, -1]] = kernel_rad_float / (2 * kernel_rad + 1)
+
+    kernel2d = torch.mm(kernel1d[:, None], kernel1d[None, :])
+
+    return kernel2d
+
+
+def gaussian_blur(img: Tensor, radius: float) -> Tensor:
+    """Performs Gaussian blurring on the img by given kernel.
+
+    Args:
+        img (Tensor): Image to be blurred
+        radius (float): Blur radius
+
+    Returns:
+        Tensor: An image that is blurred using kernel of given radius
+    """
+    if not (isinstance(img, torch.Tensor) and _is_tensor_a_torch_image(img)):
+        raise TypeError('img should be Tensor Image. Got {}'.format(type(img)))
+    if not isinstance(radius, (float, int)):
+        raise TypeError('radius should be either float or int. Got {}'.format(type(radius)))
+
+    radius = float(radius)
+    passes = 3
+
+    ndim = img.ndim
+    if ndim == 2:
+        img = img.unsqueeze(0)
+    if ndim == 3:
+        img = img.unsqueeze(0)
+
+    kernel = _get_kernel(radius, passes)
+
+    padding = _compute_padding(kernel.shape[::-1])
+
+    kernel = kernel[None, None, :, :].repeat(img.size(-3), 1, 1, 1)
+
+    padded_img = pad(img, padding, padding_mode='edge')
+    blurred_img = conv2d(padded_img, kernel, groups=img.size(-3))
+    for _ in range(passes - 1):
```

Comment:
Why do we use `passes` actually ?

Reply:
I was trying to match PIL's gaussian blur behaviour. In PIL they perform Box filter `passes` (= 3) times to approximate the required Gaussian blur.